django/views/generic/list.py

```diff
@@ -181,7 +181,13 @@ def get_template_names(self):
         if hasattr(self.object_list, 'model'):
             opts = self.object_list.model._meta
             names.append("%s/%s%s.html" % (opts.app_label, opts.model_name, self.template_name_suffix))
-
+        elif not isinstance(self.object_list, QuerySet) and not names:
```

Comment:
It looks to me like if we return an empty list from this function, then the view will definitely raise a 500 error: https://github.com/django/django/blob/master/django/template/loader.py#L49

So, the condition here can simply be `if not names:`